crates/ide-completion/src/render.rs

```diff
@@ -281,8 +281,14 @@ pub(crate) fn render_resolution_with_import(
     import_edit: LocatedImport,
 ) -> Option<Builder> {
     let resolution = ScopeDef::from(import_edit.original_item);
-    let local_name = scope_def_to_name(resolution, &ctx, &import_edit)?;
-    //this now just renders the alias text, but we need to find the aliases earlier and call this with the alias instead
+    // Use the last segment when `item_to_import` matches `original_item`,
+    // as it will take the aliased name into account.
+    let local_name = if import_edit.item_to_import == import_edit.original_item {
+        import_edit.import_path.segments().last()?.clone()
```

Comment:
I'm not sure if this is a right way to get the aliased name.

Reply:
It is a decent workaround for the core issue I mentioned for now I think, could put a FIXME comment mentioning https://github.com/rust-lang/rust-analyzer/issues/14079 as the proper fix for this above?

Reply:
I've added the FIXME.
There was a similar case with completion in pattern as well, so I added a testcase with the same fix.